packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -176,6 +179,37 @@ export class Dropdown extends BaseComponent<IDropdownProps, IDropdownState> {
     }
   }
 
+  protected _setMaxCalloutComponentHeight() {
+    this._scrollbarOffset = 0;
+    this._calloutMaxHeight = null;
+    if (this.props.maxDisplayItems && this.props.maxDisplayItems > 0) {
+      let totalItemCount = this._optionList.children.length;
+      let { selectedIndex } = this.state;
+
+      if (this.props.maxDisplayItems < totalItemCount) {
+        this._scrollbarOffset = 16;
+      }
+
+      if (selectedIndex + this.props.maxDisplayItems > totalItemCount) {
+        selectedIndex = totalItemCount - this.props.maxDisplayItems;
+      }
+
+      if (selectedIndex < 0) {
+        selectedIndex = 0;
+      }
+
+      let maxHeight = null;
+      for (let i = selectedIndex; i < this.props.maxDisplayItems + selectedIndex; i++) {
+        let liItem = this.refs[Dropdown.Option + i] as HTMLLIElement;
+        if (liItem) {
+          maxHeight += liItem.clientHeight;
+        }
+      }
+
+      this._calloutMaxHeight = maxHeight;
```

Comment:
if page size is bigger than total available item, the callout div will only use the total available's height due to <div> element's feature.

the height is calculated every time when drop down appear 

by using the selected index, the logic is:
pagesize = 3
1. A B C D E F G
       ^
then the height calc is B-D

2. A B C D E F G
                    ^
the height calc is from E-G